src/main/scala/com/webtrends/harness/component/akkahttp/routes/RouteGenerator.scala

```diff
@@ -66,22 +67,22 @@ object RouteGenerator {
                                                 responseHandler: V => Route,
                                                 errorHandler: AkkaHttpRequest => PartialFunction[Throwable, Route],
                                                 accessLogIdGetter: Option[AkkaHttpRequest => String] = None,
-                                                enableCors: Boolean = false,
-                                                defaultHeaders: Seq[HttpHeader] = Seq.empty[HttpHeader])
+                                                defaultHeaders: Seq[HttpHeader] = Seq.empty[HttpHeader],
+                                                corsSettings: Option[CorsSettings] = None)
                                                (implicit ec: ExecutionContext, log: Logger, timeout: Timeout): Route = {
 
     val httpPath = parseRouteSegments(path)
     httpPath { segments: AkkaHttpPathSegments =>
       respondWithHeaders(defaultHeaders: _*) {
-        corsSupport(method, enableCors) {
-          httpMethod(method) {
-            parameterMap { paramMap: Map[String, String] =>
-              extractRequest { request =>
-                val reqHeaders = request.headers.map(h => h.name.toLowerCase -> h.value).toMap
-                val httpEntity = getPayload(method, request)
-                val locales = requestLocales(reqHeaders)
-                val reqWrapper = AkkaHttpRequest(path, paramHoldersToList(segments), request.method, request.protocol,
-                  reqHeaders, paramMap, System.currentTimeMillis(), locales, httpEntity)
+        httpMethod(method) {

```

Comment:
Are we sure we'll still support pre-flight requests if this is before the corsSupport directive?

We also need to make sure the preflight requests work with multiple endpoints on the same path with different methods.

Reply:
No, pre-flight requests are not working when httpMethod is before the corsSupport. I have updated the code by placing httpMethod after corsSupport, now pre-flight requests are working.
Currently the pre-flight requests with multiple endpoints on the same path with different methods are not working as expected it returns only latest registered method of the endpoint, we should work on this issue. I was the one who confirmed previously it is working fine but I was tested in wrong way, sorry for that.